api/envoy/config/accesslog/v2/als.proto

```diff
@@ -60,4 +61,9 @@ message CommonGrpcAccessLogConfig {
   // this limit it hit, or every time flush interval is elapsed, whichever comes first. Setting it
   // to zero effectively disables the batching. Defaults to 16384.
   google.protobuf.UInt32Value buffer_size_bytes = 4;
+
+  // Additional filter state objects to log in :ref:`filter_state_objects
+  // <envoy_api_field_data.accesslog.v2.AccessLogCommon.filter_state_objects>`.
+  // Logger will call `FilterState::Object::serializeAsProto` to serialize the filter state object.
+  repeated string filter_state_objects_to_log = 5;
```

Comment:
Is "filter state object" the right naming terminology? Also, how does this interact with the polymorphism in `FilterState`?

Reply:
The root class for FilterState is FilterState::Object so that is the name here. I added an optional method `serializeAsProto` to that (which by default returns nullptr) to it. If a filter state is serializable they have to implement this method. It is no-op to existing FilterState other than vtable.

Reply:
The question I had is if we have two filter state objects with the same name but different types, how do we pick which one to serialize?

Reply:
filter state is a map isnt it?

Reply:
@htuch there couldn't be two filter state objects with the same name, it's a map.

Reply:
Ah, right, I thought it was a map per-type. LGTM.